.github/scripts/set-pr-status.js

```diff
@@ -0,0 +1,269 @@
+        const field = fields.find(f => f.name === fieldName);
+        if (!field) throw new Error(`Field '${fieldName}' not found`);
+        return field;
+    }
+
+    // ========== MAIN LOGIC ==========
+    const pr = context.payload.pull_request;
+
+    // Run the status check only if the PR contains the l1 label.
+    // This check is for the submitted review, the check for a pull_request event was previously done.
+    if (context.eventName === "pull_request_review") {
+        const prNumber = pr.number;
+        const { data: labels } = await github.rest.issues.listLabelsOnIssue({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            issue_number: prNumber,
+        });
+
+        const hasL1 = labels.some(label => label.name === "L1");
+        if (!hasL1) {
+            console.log("PR does not have 'L1' label. Exiting.");
+            return;
+        } else {
+            console.log("PR contains 'L1' label. Continuing with status checks...")
+        }
+    }
+
+    const projectNumber = 31;
+    const orgLogin = "lambdaclass";
+
+    // Get project and item representing Pull Request. Exiting early if it doesn't belong to the project.
+    const project = await getProject(orgLogin, projectNumber);
+    const projectId = project.id;
+
+    const allItems = await getAllProjectItems(projectId);
+    const prItem = findItemByNumber(allItems, pr.number);
+    if (!prItem) {
+        console.warn(`PR #${pr.number} not found in ethrex_l1 project. Exiting...`);
+        return;
+    }
+
+    // Get all fields of the project.
+    const fields = await getProjectFields(projectId);
+
+    const statusField = getFieldByName(fields, "Status");
+    const lastUpdatedField = getFieldByName(fields, "Last updated");
+    const startDateField = getFieldByName(fields, "Start date");
+
+    const action = context.payload.action;
+
+    // Set Start date if PR was opened
+    if (action === "opened") {
+        await setProjectItemDateField(projectId, prItem.id, startDateField, pr.created_at);
+    }
+
+    // Set date of Last update only if it's an update to the PR
+    if (action === "synchronize" || action === "edited") {
+        await setProjectItemDateField(projectId, prItem.id, lastUpdatedField, pr.updated_at);
+    }
+
+    const isDraftTransition = (action === "opened" || action === "reopened") && pr.draft || action === "converted_to_draft";
+    const readyForReview = action === "ready_for_review";
+    const requestedChanges = context.eventName === "pull_request_review" && context.payload.review?.state === "changes_requested";
+
+    // Determine new status of the PR.
+    let newStatus = null;
+
+    if (isDraftTransition) {
+        newStatus = "In Progress";
+    } else if (readyForReview || ((action === "opened" || action === "reopened") && !pr.draft)) {
+        newStatus = "In Review";
+    } else if (requestedChanges) {
+        newStatus = "Requested Changes";
```

Comment:
I wouldn't add a new status called "Requested Changes". Instead I would move it to "In Progress" again. I don't think it's worth it to add a new status for this.